src/tdycoremesh.c

```diff
@@ -0,0 +1,1230 @@
+            break;
+          }
+        }
+        c2vCount++;
+      } else if (IsClosureWithinBounds(closure[i], eStart,
+                                       eEnd)) { /* Is the closure an edge? */
+        iedge = closure[i] - eStart;
+        cells[icell].edge_ids[c2eCount] = iedge;
+        for (j=0; j<2; j++) {
+          if (edges[iedge].cell_ids[j] == -1) {
+            edges[iedge].cell_ids[j] = icell;
+            break;
+          }
+        }
+        c2eCount++;
+      }
+    }
+  }
+
+  // edge--to--vertex
+  for (e=eStart; e<eEnd; e++) {
+    ierr = DMPlexGetConeSize(dm, e, &coneSize); CHKERRQ(ierr);
+    ierr = DMPlexGetCone(dm, e, &cone); CHKERRQ(ierr);
+    ierr = DMPlexGetSupportSize(dm, e, &supportSize); CHKERRQ(ierr);
+    iedge = e-eStart;
+
+    if (supportSize == 1) edges[iedge].is_internal = PETSC_FALSE;
+    else                  edges[iedge].is_internal = PETSC_TRUE;
+
+    edges[iedge].vertex_ids[0] = cone[0]-vStart;
+    edges[iedge].vertex_ids[1] = cone[1]-vStart;
+
+    for (d=0; d<dim; d++) {
+      v_1[d] = vertices[edges[iedge].vertex_ids[0]].coordinate.X[d];
+      v_2[d] = vertices[edges[iedge].vertex_ids[1]].coordinate.X[d];
+    }
+    ierr = ComputeLength(v_1, v_2, dim, &(edges[iedge].length)); CHKERRQ(ierr);
+  }
+
+  // vertex--to--edge
+  for (v=vStart; v<vEnd; v++) {
+    ierr = DMPlexGetSupport(dm, v, &support); CHKERRQ(ierr);
+    ierr = DMPlexGetSupportSize(dm, v, &supportSize); CHKERRQ(ierr);
+    ivertex = v - vStart;
+    vertices[ivertex].num_edges = supportSize;
+    for (s=0; s<supportSize; s++) {
+      iedge = support[s] - eStart;
+      vertices[ivertex].edge_ids[s] = iedge;
+      if (!edges[iedge].is_internal) vertices[ivertex].num_boundary_cells++;
+    }
+  }
+
+  PetscFunctionReturn(0);
+
+}
+
+/* -------------------------------------------------------------------------- */
+
+PetscErrorCode ComputeTheta(PetscReal x, PetscReal y, PetscReal *theta) {
+
+  PetscFunctionBegin;
+
+  if (x>0.0) {
+    if (y>= 0.0) *theta = atan(y/x);
+    else         *theta = atan(y/x) + 2.0*PETSC_PI;
+  } else if (x==0.0) {
+    if      (y>  0.0) *theta = 0.5*PETSC_PI;
+    else if (y ==0.0) *theta = 0.;
+    else              *theta = 1.5*PETSC_PI;
+  } else {
+    *theta = atan(y/x) + PETSC_PI;
+  }
+
+  PetscFunctionReturn(0);
+}
+
+/* -------------------------------------------------------------------------- */
+
+PetscErrorCode UpdateCellOrientationAroundAVertex(TDy tdy, PetscInt ivertex) {
+
+  PetscFunctionBegin;
+
+  TDy_mesh       *mesh;
+  TDy_cell       *cells;
+  TDy_vertex     *vertex;
+  TDy_edge       *edges;
+  PetscInt       icell, iedge;
+  PetscInt       ncells, nedges;
+  PetscReal      x,y;
+  PetscReal      theta[8];
+  PetscInt       idx[8], ids[8], subcell_ids[8];
+  PetscInt       tmp_subcell_ids[4];
+  PetscInt       tmp_cell_ids[4], tmp_edge_ids[4], tmp_ncells, tmp_nedges;
+  PetscInt       count;
+  PetscInt       start_idx;
```

Comment:
```suggestion
  PetscInt       start_idx=-1;
```
I am not sure what to initialize this to, but because it first gets set in a conditional statement, the compiler complains.

Reply:
Fixed by https://github.com/TDycores-Project/TDycore/pull/5/commits/e7441c7e94d399ab7f76cee3617631efa9a1ae3a#diff-fcf52e8a4105ab2e05e6a4f90c5dfbd9R555